### PR TITLE
feat(console): add tenant env migration modal

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

Add tenant env migration modal according to the migration data (implemented in https://github.com/logto-io/cloud/pull/338)

### Updates
Move the `TenantTag` type into its own type file, or the `logto-config` type will rely on the original `TenantTag` from tenant model generated file, this will cause an module build error:

```
Cannot find module '@withtyped/server/model' from '../schemas/lib/models/tenants.js'
```

image

## Testing

image

image

image

## Checklist

~~- [ ] `.changeset`~~
~~- [ ] unit tests~~
~~- [ ] integration tests~~
~~- [ ] necessary TSDoc comments~~

### CODE REVIEW

File: packages/console/src/containers/AppContent/TenantNotificationContainer/TenantEnvMigrationModal/index.module.scss [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  top: 0;
+  pointer-events: none;
+
+  .fireworksImage {
+    transform: translateX(-10.5%) translateY(-17%);
+  }
+
+  .stagingFireworksImage {
+    transform: translateX(-10.5%) translateY(-38%);
+  }
```
Comment (charIeszhao) [inline packages/console/src/containers/AppContent/TenantNotificationContainer/TenantEnvMigrationModal/index.module.scss:37]: Would this transform work in different screen sizes?

File: packages/console/src/containers/AppContent/TenantNotificationContainer/TenantEnvMigrationModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+import ModalLayout from '@/ds-components/ModalLayout';
+import useConfigs from '@/hooks/use-configs';
+import useTheme from '@/hooks/use-theme';
+import * as modalStyles from '@/scss/modal.module.scss';
+
+import DevelopmentTenantFeatures from './DevelopmentTenantFeatures';
+import DevelopmentTenantMigrationHint from './DevelopmentTenantMigrationHint';
+import * as styles from './index.module.scss';
+
+const shouldDevelopmentTenantFeatureList = (originalTenantTag: TenantTag, isPaidTenant: boolean) =>
```
Comment (wangsijie) [inline packages/console/src/containers/AppContent/TenantNotificationContainer/TenantEnvMigrationModal/index.tsx:22]: Is there a typo?

**Review by wangsijie — APPROVED** [review]

**Review by charIeszhao — APPROVED** [review]

File: packages/console/src/containers/AppContent/TenantNotificationContainer/TenantEnvMigrationModal/index.module.scss (hunk 1 above)
Comment (xiaoyijun) [reply packages/console/src/containers/AppContent/TenantNotificationContainer/TenantEnvMigrationModal/index.module.scss:37]: Yes, tested.

File: packages/console/src/containers/AppContent/TenantNotificationContainer/TenantEnvMigrationModal/index.tsx (hunk 2 above)
Comment (xiaoyijun) [reply packages/console/src/containers/AppContent/TenantNotificationContainer/TenantEnvMigrationModal/index.tsx]: Updated.

**Pull request merged**